apps-client/backoffice/app/routes/activity.internal-election._index/data.tsx

```diff
@@ -0,0 +1,293 @@
+                if (!data) return
+                const idx = data.data.findIndex((d) => d.id === electionId)
+                if (idx === -1) return
+                data.data[idx].isCancelled = true
+                return data
+              }
+            )
+          },
+        }
+      )
+    },
+    [cancelMutation, queryClient, queryLimit, queryPage, queryName, queryStatus]
+  )
+
+  const deleteElection = useCallback(
+    (electionId: string) => {
+      if (deleteMutation.isPending) return
+
+      deleteMutation.mutateAsync(
+        { pathParams: { electionId } },
+        {
+          onSuccess: () => {
+            queryClient.setQueryData(
+              reactQueryClient.getQueryKey('/admin/elections', {
+                query: {
+                  limit: queryLimit,
+                  page: queryPage,
+                  name: queryName,
+                  status: queryStatus,
+                },
+              }),
+              (_data) => {
+                const data = structuredClone(_data)
+                if (!data) return
+                data.data = data.data.filter((d) => d.id !== electionId)
+                return data
+              }
+            )
+          },
+        }
+      )
+    },
+    [deleteMutation, queryClient, queryLimit, queryPage, queryName, queryStatus]
+  )
+
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor('id', {
+        header: () => <div className="pl-2">ID</div>,
+        cell: (info) => <TableCopyId id={info.getValue()} />,
+        size: 64,
+        minSize: 64,
+        maxSize: 64,
+      }),
+      columnHelper.accessor('name', {
+        header: 'ชื่อการเลือกตั้ง',
+        cell: (info) => (
+          <NavLink
+            className="hover:underline"
+            to={`/activity/internal-election/${info.row.original.id}`}
+          >
+            {info.getValue()}
+          </NavLink>
+        ),
+      }),
+      columnHelper.accessor('district', {
+        header: 'อำเภอ/เขต',
+        cell: (info) => info.getValue() ?? '-',
+        size: 110,
+        minSize: 110,
+      }),
+      columnHelper.accessor('province', {
+        header: 'จังหวัด',
+        cell: (info) => info.getValue() ?? '-',
+        size: 110,
+        minSize: 110,
+      }),
+      columnHelper.accessor('status', {
+        header: 'สถานะ',
+        cell: (info) => <ElectionStatusBadge status={info.getValue()} />,
+        size: 110,
+        minSize: 110,
+      }),
+      columnHelper.display({
+        id: 'votingTimeline',
+        header: 'ช่วงเวาลงคะเเนน',
+        cell: (info) => {
+          const openVote = new Date(info.row.original.openVoting).toLocaleDateString('en-GB', {
+            day: '2-digit',
+            month: '2-digit',
+            year: '2-digit',
+          })
+
+          const closeVote = new Date(info.row.original.closeVoting).toLocaleDateString('en-GB', {
+            day: '2-digit',
+            month: '2-digit',
+            year: '2-digit',
+          })
```

Comment:
suggestion: use `th` locale for date

> note to @rootEnginear: maybe extract this into date utils.

```suggestion
          const openVote = new Date(info.row.original.openVoting).toLocaleDateString('th', {
            day: '2-digit',
            month: '2-digit',
            year: '2-digit',
          })

          const closeVote = new Date(info.row.original.closeVoting).toLocaleDateString('th', {
            day: '2-digit',
            month: '2-digit',
            year: '2-digit',
          })
```